### PR TITLE
feat(git-version): Add version to the UI and config endpoint

### PR DESCRIPTION
**Pull request by dexter-mh-lee**
Add version to the UI and config endpoint by
1: Add git properties plugin to all java based gradle tasks that generates a git.properties file with the commit hash and other necessary information (latest tag for the given commit)
2: Add GitVersionFactory that fetches the commit hash and version from the git.properties
3: Add the version info to the config endpoint

Also modified all github actions that checkout code to build image to fetch all history (otherwise git describe does not work)
```
curl http://localhost:8080/config

{
  "models" : { },
  "versions" : {
    "linkedin/datahub" : {
      "version" : "v0.8.22",
      "commit" : "22acafafa4b86c1ee9623bc6023851a50829949d"
    }
  },
  "statefulIngestionCapable" : true,
  "retention" : "true",
  "noCode" : "true"
}
```
4: Add the version info in the UI
Screen Shot 2022-01-10 at 9 35 14 PM

## Checklist
- [ ] The PR conforms to DataHub's [Contributing Guideline](https://github.com/linkedin/datahub/blob/master/docs/CONTRIBUTING.md) (particularly [Commit Message Format](https://github.com/linkedin/datahub/blob/master/docs/CONTRIBUTING.md#commit-message-format))
- [ ] Links to related issues (if applicable)
- [ ] Tests for the changes have been added/updated (if applicable)
- [ ] Docs related to the changes have been added/updated (if applicable)

### CODE REVIEW

File: datahub-web-react/src/app/shared/ManageAccount.tsx [hunk 1]
```diff
@@ -44,14 +45,16 @@ const defaultProps = {
...
     const handleLogout = () => {
         analytics.event({ type: EventType.LogOutEvent });
         isLoggedInVar(false);
         Cookies.remove(GlobalCfg.CLIENT_AUTH_COOKIE);
     };
-
+    const version = config?.appVersion;
     const menu = (
         <Menu>
+            {version && <MenuItem key="version">{version}</MenuItem>}
```
Comment (gabe-lyons) [inline datahub-web-react/src/app/shared/ManageAccount.tsx:57]: can we make this link to release notes 🥺

**Review by gabe-lyons — APPROVED** [review]
Love it!

**Review by shirshanka — APPROVED** [review]
LGTM!

File: datahub-web-react/src/app/shared/ManageAccount.tsx (hunk 1 above)
Comment (jjoyce0510) [reply datahub-web-react/src/app/shared/ManageAccount.tsx:57]: @dexter-mh-lee We can wrap this in a <Link to="url-to-release-notes"> component to do this^. Me or Gabe can also do it in a followup if its easier.

File: metadata-io/src/main/java/com/linkedin/metadata/version/GitVersion.java [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+
+@Value
+public class GitVersion {
+  String version;
+  String commitId;
+  Optional<String> flag;
+
+  public static GitVersion getVersion(@Nonnull String commitId, @Nonnull String commitDescribe) {
+    String version = commitDescribe.split("-")[0];
+    return new GitVersion(version, commitId, Optional.empty());
```
Comment (jjoyce0510) [inline metadata-io/src/main/java/com/linkedin/metadata/version/GitVersion.java:18]: Does this get the first tag?

**Pull request merged**